bingart/cli.py

```
@@ -0,0 +1,325 @@
+        sys.exit(EXIT_GENERIC_ERROR)
+    if not cookie:
+        print("Error: no cookie provided.", file=sys.stderr)
+        sys.exit(EXIT_AUTH_ERROR)
+    return cookie, False
+
+
+async def download_file(url, dest_path, session=None):
+    import urllib.request
+
+    try:
+        urllib.request.urlretrieve(url, str(dest_path))
```

Ruff (0.15.9)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)